### PR TITLE
Run wasm console sample on Helix

### PR DESCRIPTION
**Pull request by MaximLipnin**
- Included the WASM console sample project to the library test build. The sample can still be built/run locally using make file.
- Added zipping/copying the sample application to helix directory from where it can be consumed by CI.
- To run on Helix, tried to use a simple helix work item which wouldn't rely on generated RunTests.sh and just call the xharness command directly.  

Relates to https://github.com/dotnet/runtime/issues/43865

### CODE REVIEW

**Comment by radical** [comment]
@MaximLipnin does the sample run on helix now? Can you share a link to the log? I wasn't able to get to it.

**Comment by MaximLipnin** [comment]
@radical yeah, it should, please take a look at logs:

https://helix.dot.net/api/jobs/d971e44a-7a7c-42b5-92cf-36565dc3a4c6/workitems/WasmConsoleSample?api-version=2019-06-17

https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-45768-merge-d971e44a7a7c42b592/WasmConsoleSample/console.56eedcdd.log?sv=2019-07-07&se=2021-01-04T18%3A46%3A00Z&sr=c&sp=rl&sig=fqaYZNVFNghjNifRWWmDVXyeRYmwFOOSqkF6SAz8lGc%3D

**Review by radical — APPROVED** [review]
@akoeplinger would be able to review this well. I don't know enough about the helix related bits.

- I'm wondering if we can/should be generating `RunTests.sh`.
- We could make this handle more samples, or similar projects too, but that doesn't have to be in this PR.
- In general, this looks fine, and it seems to work!
- 
- It should be useful to get this in anyway, as it will prevent samples getting unintentionally broken.

**Review by akoeplinger — APPROVED** [review]

**Comment by akoeplinger** [comment]
> I'm wondering if we can/should be generating RunTests.sh

That is only done by the libraries tests afaik.

File: src/libraries/tests.proj [hunk 1]
```diff
@@ -175,6 +175,9 @@
     <ProjectReference Include="..\mono\netcore\sample\iOS\Program.csproj"
                       BuildInParallel="false"
                       Condition="'$(ArchiveTests)' == 'true' and ('$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'tvOS')" />
+    <ProjectReference Include="..\mono\netcore\sample\wasm\console\WasmSample.csproj"
```
Comment (akoeplinger) [inline src/libraries/tests.proj:178]: missed this earlier, but @MaximLipnin can you please change this and the above to use

```suggestion
    <ProjectReference Include="$(MonoProjectRoot)netcore\sample\wasm\console\WasmSample.csproj"
```

Comment (MaximLipnin) [reply src/libraries/tests.proj:178]: Yeah, thanks, I wanted to fix it during all the previous PRs but missed as well.

Comment (MaximLipnin) [reply src/libraries/tests.proj:178]: Updated in https://github.com/dotnet/runtime/pull/46129

**Comment by safern** [comment]
> That is only done by the libraries tests afaik.

This is done here. You could technically import these targets for the sample project and just hook them up: https://github.com/dotnet/runtime/blob/93e767bd18975bf33d64177f6849c12d7505d206/eng/testing/tests.targets#L51-L52

I think I would rather do that and try to be consistent, it seems cumbersome to have that "workaround" in sendhelix to include the sample as part of the payload and hardcode the `HelixCommand`.

**Pull request merged**